app/Traits/CallsBucketsTrait.php

```diff
@@ -134,4 +146,10 @@ private function encryptValues($string_to_sign, $service, $region = 'us-east-1')
 
         return $signature;
     }
+
+    /**
+     * abstract method to get the API key
+     *
+     */
+    abstract protected function getKey();
```

Comment:
This abstract method has been created to have access to the key value from this trait. So, the controllers that use this trait must to implement this method. Currently, the controllers extend from the APIController.php class and I have implemented the getKey method on this class.